src/standalone/Vegetation/solar_induced_fluorescence.jl

```diff
@@ -34,10 +34,12 @@ Base.eltype(::SIFParameters{FT}) where {FT} = FT
 
 struct Lee2015SIFModel{FT, SP <: SIFParameters{FT}} <: AbstractSIFModel{FT}
     parameters::SP
-    function Lee2015SIFModel{FT}() where {FT}
-        parameters = SIFParameters{FT}()
-        new{FT, typeof(parameters)}(parameters)
-    end
+end
+
+function Lee2015SIFModel{FT}(
+    parameters = SIFParameters{FT},
+   ) where {FT <: AbstractFloat}
+    return Lee2015SIFModel{eltype(parameters), typeof(parameters)}(parameters)
```

Comment:
but Im not sure you can have a default keyword arg with parametric FT, because FT is not known until you call the function. but to call the function and use the that default, you need FT?

```
julia> function foo(a = FT(2)) where {FT}
       return a
       end
WARNING: method definition for foo at REPL[1]:1 declares type variable FT but does not use it.
foo (generic function with 2 methods)

julia> foo()
ERROR: UndefVarError: `FT` not defined
Stacktrace:
 [1] foo()
   @ Main ./REPL[1]:2
 [2] top-level scope
   @ REPL[2]:1

```

Reply:
nvm , it seems OK
```
julia> struct Tmp2{FT<:AbstractFloat}
       a::FT
       end

julia> function Tmp2{FT}(a = FT(2)) where {FT}
       return Tmp2{FT}(a)
       end

julia> Tmp2{Float64}()
Tmp2{Float64}(2.0)
```